### PR TITLE
Parse XDG_CURENT_DESKTOP as list

### PR DESCRIPTION
**Pull request by columbarius**
Also fixes memory leak in get_config_path

### CODE REVIEW

File: src/core/config.c [hunk 1]
```diff
@@ -115,30 +115,56 @@ static char *get_config_path(void) {
...
+		xdg_current_desktop = strdup(xdg_current_desktop);
+	}
+
+	const char *config[config_num];
+	if (xdg_current_desktop != NULL) {
+		char *token = strtok(xdg_current_desktop, ":");
+		for (int i = 0; i < config_num - 1; i++) {
+			logprint(TRACE, "xdpw: token: %s (%p)", token, token);
+			config[i] = token;
+			token = strtok(NULL, ":");
```
Comment (columbarius) [inline src/core/config.c:138]: 1. Is strtok nilpotent (safe to call arbitrary often)?
2. Why is strtok "skipping" delimiters?
    ```
    % XDG_CURRENT_DESKTOP="::sway:wlroots" ./build/xdg-desktop-portal-wlr -l TRACE  
    2021/04/23 01:12:16 [TRACE] - xdpw: token: sway (0x602000000010)
    2021/04/23 01:12:16 [TRACE] - xdpw: token: wlroots (0x602000000015)
    2021/04/23 01:12:16 [TRACE] - xdpw: token: (null) ((nil))
    2021/04/23 01:12:16 [TRACE] - xdpw: token: (null) ((nil))
    2021/04/23 01:12:16 [TRACE] - config: trying config file ~/.config/xdg-desktop-portal-wlr/sway
    2021/04/23 01:12:16 [TRACE] - config: trying config file ~/.config/xdg-desktop-portal-wlr/wlroots
    2021/04/23 01:12:16 [TRACE] - config: trying config file ~/.config/xdg-desktop-portal-wlr/config
    2021/04/23 01:12:16 [TRACE] - config: trying config file /usr/local/etc/xdg/xdg-desktop-portal-wlr/sway
    2021/04/23 01:12:16 [TRACE] - config: trying config file /usr/local/etc/xdg/xdg-desktop-portal-wlr/wlroots
    2021/04/23 01:12:16 [TRACE] - config: trying config file /usr/local/etc/xdg/xdg-desktop-portal-wlr/config
    ```

Comment (columbarius) [reply src/core/config.c:138]: 1. should be according to the manpage
2. default behaviour according to the manpage

**Comment by columbarius** [comment]
... Changed the title, but didn't remove the draft setting

**Comment by columbarius** [comment]
Rebased and "simplified" on the cost of one inline if.

File: src/core/config.c [hunk 2]
```diff
@@ -119,24 +119,46 @@ static char *get_config_path(void) {
...
+	if (xdg_current_desktop && xdg_current_desktop[0] != '\0') {
+		config_num++;
+		for (size_t i = 0; xdg_current_desktop[i] != '\0'; i++) {
+			if (xdg_current_desktop[i] == ':')
+				config_num++;
+		}
+	}
+	xdg_current_desktop = config_num > 1 ? strdup(xdg_current_desktop) : NULL;
+
+	const char *config[config_num];
```
Comment (emersion) [inline src/core/config.c:134]: I don't think we actually need to build this list? Can't we just call `strtok` unconditionally in the loop that calls `config_path`?

Comment (columbarius) [reply src/core/config.c:134]: we would have to strdup getenv("XDG_CURRENT_DESKTOP") for every prefix in the loop, but that should also work

Comment (emersion) [reply src/core/config.c:134]: Maybe swap the outer and inner loops to avoid this?

Comment (columbarius) [reply src/core/config.c:134]: then we would alternate between user and system config. I would prefere to first check all possible user configs and then the system ones.

**Comment by danshick** [comment]
Apologies for the delay, I can test this tomorrow.

**Review by danshick — CHANGES_REQUESTED** [review]
Do we want to update the documentation [here](https://github.com/emersion/xdg-desktop-portal-wlr/blame/master/xdg-desktop-portal-wlr.5.scd#L13-L20) to reflect this change? Add a line like, "$XDG_CURRNET_DESKTOP can be a colon-separated list which we will process in order".

Otherwise, this runs fine and the implementation looks good.

**Review by danshick — APPROVED** [review]

File: src/core/config.c (hunk 2 above)
Comment (columbarius) [reply src/core/config.c]: I changed it to that. It looks a little bit cleaner. Choose which one you prefer and we can use that version

**Comment by danshick** [comment]
I strongly prefer this approach, even though it is a little more verbose. I'll test later today or tomorrow and we can merge.

**Comment by columbarius** [comment]
Squashed and ready for testing and merging

**Comment by columbarius** [comment]
Please don't merge it, when you finished testing. The last commitmessage has to be reworded s/README/doc/.

**Review by danshick — APPROVED** [review]

**Comment by danshick** [comment]
Looks good to me. Please go ahead and fixup your commit message, force push, and merge.

File: src/core/config.c [hunk 3]
```diff
@@ -119,24 +119,44 @@ static char *get_config_path(void) {
 	prefix[0] = config_home;
 	prefix[1] = SYSCONFDIR "/xdg";
 
-	const char *config[2];
-	config[0] = getenv("XDG_CURRENT_DESKTOP");
-	config[1] = "config";
+	char *xdg_current_desktop = getenv("XDG_CURRENT_DESKTOP");
+	char *config_fallback = "config";
```
Comment (emersion) [inline src/core/config.c:123]: Nit: these two can be `const`

**Review by emersion — COMMENTED** [review]
Feel free to merge!

File: src/core/config.c (hunk 3 above)
Comment (columbarius) [reply src/core/config.c]: done

**Comment by columbarius** [comment]
Thanks for the review!

**Pull request merged**